### PR TITLE
android: make IPNService a foreground service

### PR DESCRIPTION
**Pull request by oxtoacart**
Prevents BackgroundServiceStartNotAllowedException.

Updates #cleanup

### CODE REVIEW

File: android/src/main/java/com/tailscale/ipn/App.kt [hunk 1]
```diff
@@ -441,6 +415,30 @@ open class UninitializedApp : Application() {
...
+        PendingIntent.getBroadcast(
+            this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+    return NotificationCompat.Builder(this, STATUS_CHANNEL_ID)
+        .setSmallIcon(icon)
+        .setContentTitle("Tailscale")
+        .setContentText(message)
+        .setAutoCancel(!vpnRunning)
+        .setOnlyAlertOnce(!vpnRunning)
+        .setOngoing(vpnRunning)
+        .setSilent(true)
```
Comment (oxtoacart) [inline android/src/main/java/com/tailscale/ipn/App.kt:439]: Drive-by, make this silent.

File: android/src/main/java/com/tailscale/ipn/App.kt [hunk 2]
```diff
@@ -441,6 +415,30 @@ open class UninitializedApp : Application() {
...
+        .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+        .addAction(NotificationCompat.Action.Builder(0, actionLabel, pendingIntent).build())
```
Comment (oxtoacart) [inline android/src/main/java/com/tailscale/ipn/App.kt:441]: Drive-by, linter didn't like us using a content intent, wants us to use actions instead.

**Review by agottardo — APPROVED** [review]

**Comment by bradfitz** [comment]
See http://go/cleanup :)

BackgroundServiceStartNotAllowedException sounds not good... was this crashing?

File: android/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -10,6 +10,8 @@
         android:maxSdkVersion="29" />
     <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
```
Comment (bradfitz) [inline android/src/main/AndroidManifest.xml:14]: https://developer.android.com/about/versions/14/changes/fgs-types-required says:

> In addition to declaring the FOREGROUND_SERVICE_TYPE_SPECIAL_USE foreground service type, developers should declare use cases in the manifest. To do so, they specify the <property> element within the <service> element. These values and corresponding use cases are reviewed when you submit your app in the Google Play Console. The use cases you provide are free-form, and you should make sure to provide enough information to let the reviewer see why you need to use the specialUse type.

Will this get rejected by the play store without a reason?

Comment (agottardo) [reply android/src/main/AndroidManifest.xml:14]: Sounds like we should use https://developer.android.com/about/versions/14/changes/fgs-types-required#system-exempted given it specifically mentions VPN apps as a use-case.

**Review by kari-ts — APPROVED** [review]
